js/src/controls/SearchControl.js

```diff
@@ -32,18 +32,4 @@ export class LeafletSearchControlView extends control.LeafletControlView{
   create_obj() {
     this.obj = L.control.search(this.get_options());
```

Comment:
You did not tell to ipyleaflet that the `marker` attribute was an option Python side (the tag `o=True` is missing), but I guess this is fine. Only the marker won't be part of the `get_options` output here, so you need to integrate it in some way. Something like:

```javascript
this.create_child_view(this.model.get('marker')).then((view) => {
  let options = this.get_options();
  options.marker = view;
  this.obj = L.control.search(options);
});
```

Reply:
Again you can check the Popup implementation for the child view creation.

Reply:
Done.

Reply:
I have removed child view creation logic as it throws below exception.
```
leaflet-src.js:4756 Uncaught (in promise) TypeError: Cannot read property 'addTo' of undefined
    at e.addControl (leaflet-src.js:4756)
    at Map.js:160
    at async Promise.all (index 2)
```
It works fine without a child view for the marker. let me know your thoughts.

Reply:
Yes, I feel like it's because you need to return the promise in the method, something like:

```javascript
return this.create_child_view(this.model.get('marker')).then((view) => {
  let options = this.get_options();
  options.marker = view;
  this.obj = L.control.search(options);
});
```

I'll test and merge your PR tomorrow or this week-end :)

Reply:
Thanks.